programs/quarter_board/main.c

```diff
@@ -0,0 +1,182 @@
+#include "ch.h"
+#include "hal.h"
+
+#include <icu_lld.h>
+#include "chprintf.h"
+#include <inttypes.h>
+
+#include "communication.h"
+
+#define MAX_PULSE_LENGTH 3000
+#define MIN_PULSE_LENGTH 400
+#define NOT_SEEING_VALUE 3000
+#define QUANTITY_OF_SENSORS 2
+
+#define UNUSED(arg) (void)(arg)
+
+int32_t sensors_values[QUANTITY_OF_SENSORS];
+
+thread_t *commanderp;
+thread_t *ir_measurementp;
+
+static void read_pulse(ICUDriver *icup) {
+    uint8_t i;
+    int32_t value;
+
+    if (icup == &ICUD1) {
+        i = 0;
+    } else {
+        i = 1;
+    }
+    
+    value = icuGetWidthX(icup);
+
+    if (value < MAX_PULSE_LENGTH && value > MIN_PULSE_LENGTH) {    
+        sensors_values[i] = value;
+    } else {
+        sensors_values[i] = NOT_SEEING_VALUE;
+    }
+}
+
+void execute_master_command(uint16_t command_id, uint8_t *buff) {
+    switch (command_id) {
+        case 0xAA:  
+            if(buff[0] == 1) {

```

Comment:
Is my guess that `buff[0]` and `buff[1]` can only contain 1 or 0? Because if so, why not use **that** as value for `led` and avoiding these `if`-s? And even if not, why not use: `led(GREEN, (buff[0] == 1));` ?